File: app/models/enumeration/SearchType.java
package models.enumeration;

/**
 * @author Keeun Baik
 */
public enum SearchType {

    AUTO("auto"), NA("not available"), USER("user"), PROJECT("project"), ISSUE("issue"), POST("post"),
    MILESTONE("milestone"), ISSUE_COMMENT("issue_comment"), POST_COMMENT("post_comment"), REVIEW("review");

    private String searchType;

    SearchType(String value) {
        this.searchType = value;
    }

    public String getType() {
        return this.searchType;
    }

    public static SearchType getValue(String value) {
        for (SearchType searchType : SearchType.values()) {
            if (searchType.searchType.equals(value)) {
                return searchType;
            }
        }
        return NA;
    }

}

File: src/main/kotlin/com/github/yonaprojects/yona/domain/enumeration/SearchType.kt
package com.github.yonaprojects.yona.domain.enumeration

enum class SearchType(val type: String) {
    AUTO("auto"),
    NA("not available"),
    USER("user"),
    PROJECT("project"),
    ISSUE("issue"),
    POST("post"),
    MILESTONE("milestone"),
    ISSUE_COMMENT("issue_comment"),
    POST_COMMENT("post_comment"),
    REVIEW("review"),
    // `yona search prs` 대응. PR 자체를 색인하는 통합검색 SearchType이 원래 없었다.
    PULL_REQUEST("pull_request");

    companion object {
        fun getValue(value: String): SearchType {
            return values().find { it.type == value } ?: NA
        }
    }
}
